src/main/java/org/opensearch/security/privileges/PitAccessEvaluator.java

```diff
@@ -0,0 +1,344 @@
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.security.OpenSearchSecurityPlugin;
+import org.opensearch.security.resolver.IndexResolverReplacer;
+import org.opensearch.security.securityconf.SecurityRoles;
+import org.opensearch.security.user.User;
+
+/**
+ * This class evaluates privileges for point in time (Delete and List all) operations
+ */
+public class PitAccessEvaluator {
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+    private boolean isDebugEnabled = log.isDebugEnabled();
+
+    public PrivilegesEvaluatorResponse evaluate(final ActionRequest request, final ClusterService clusterService,
+                                                final User user, final SecurityRoles securityRoles, final String action,
+                                                final IndexNameExpressionResolver resolver,
+                                                final PrivilegesEvaluatorResponse presponse) {
+
+        // Skip pit evaluation for "NodesGetAllPITs" action, since it fetches all PITs across the cluster
+        // for privilege evaluation
+        if(action.startsWith("cluster:admin")) {
+            return presponse;
+        }
+        try {
+            if (request instanceof GetAllPitNodesRequest) {
+                if (((GetAllPitNodesRequest) request).getGetAllPitNodesResponse() != null) {
+                    return presponse;
+                }
+                return handleGetAllPitsAccess(request, clusterService, user, securityRoles,
+                        action, resolver, presponse);
+            } else if (request instanceof DeletePitRequest) {
+                DeletePitRequest deletePitRequest = (DeletePitRequest) request;
+                List<String> pitIds = deletePitRequest.getPitIds();
+                if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+                    return handleDeleteAllPitAccess(deletePitRequest, clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                } else {
+                    return handleExplicitPitsAccess(deletePitRequest.getPitIds(), clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                }
+            } else if (request instanceof PitSegmentsRequest) {
+                PitSegmentsRequest pitSegmentsRequest = (PitSegmentsRequest) request;
+                List<String> pitIds = pitSegmentsRequest.getPitIds();
+                if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+                    return handleGetAllPitSegmentsAccess(pitSegmentsRequest, clusterService, user, securityRoles,
+                            action, resolver, presponse);
+                } else {
+                    return handleExplicitPitsAccess(pitSegmentsRequest.getPitIds(), clusterService, user, securityRoles,
```

Comment:
Can `deletePitRequest.getPitIds()` be replaced with `pitIds` here to avoid unnecessary performance overhead?